pkg/adminapi/client.go

```diff
@@ -71,3 +74,33 @@ func MakeHTTPClient(opts *HTTPClientOpts) (*http.Client, error) {
 
 	return c, nil
 }
+
+// GetKongClientForWorkspace returns a Kong API client for a given root API URL and workspace.
+// If the workspace does not already exist, GetKongClientForWorkspace will create it.
+func GetKongClientForWorkspace(ctx context.Context, adminURL string, wsName string,
+	httpclient *http.Client) (*kong.Client, error) {
+	client, err := kong.NewClient(kong.String(adminURL), httpclient)
+	if err != nil {
+		return nil, fmt.Errorf("creating Kong client: %w", err)
+	}
+	if wsName != "" {
```

Comment:
If you invert this expression, there will be less indentation in this function, which improves code readability:

```go
if wsName == "" {
  // do the simple 1-liner thing
}

// do the complicated lengthy thing 1 level of nesting higher

```